src/time.rs

```diff
@@ -3,7 +3,8 @@ use {Result, Errno};
 use libc;
 
 libc_enum! {
-    #[repr(i32)]
+    #[cfg_attr(target_os = "darwin", repr(i32))]
+    #[cfg_attr(not(target_os = "darwin"), repr(u32))]
```

Comment:
This isn't correct.  Check the libc crate to see what the actual size is on various platforms:
```
> rg type.clockid_t
linux_like/mod.rs
4:pub type clockid_t = ::c_int;

newlib/mod.rs
4:pub type clockid_t = ::c_ulong;

uclibc/mod.rs
6:pub type clockid_t = ::c_int;

solarish/mod.rs
5:pub type clockid_t = ::c_int;

haiku/mod.rs
9:pub type clockid_t = i32;

redox/mod.rs
9:pub type clockid_t = ::c_int;

hermit/mod.rs
39:pub type clockid_t = c_ulong;

bsd/netbsdlike/mod.rs
10:pub type clockid_t = ::c_int;

bsd/apple/mod.rs
16:pub type clockid_t = ::c_uint;

bsd/freebsdlike/dragonfly/mod.rs
8:pub type clockid_t = ::c_ulong;

bsd/freebsdlike/freebsd/mod.rs
6:pub type clockid_t = ::c_int;
```